src/main/java/com/jorgepatrick/NumberConverter.java

```diff
@@ -72,25 +82,41 @@ public int parseRomanToArabic(final String romanNumber) {
 
         romanNumberValidator.validateRomanNumber(romanNumber);
 
-        int unit = 0;
+        int arabicNumber = 0;
 
         for (int currentDigit = 0; currentDigit < romanNumber.length(); currentDigit++) {
-            if (romanNumber.toUpperCase().charAt(currentDigit) == I.value().charAt(0)) {
-                int nextDigit = currentDigit + 1;
-                if (nextDigit == romanNumber.length() ||
-                    romanNumber.toUpperCase().charAt(nextDigit) == I.value().charAt(0)) {
-                    unit += 1;
-                } else {
-                    unit -= 1;
-                }
+
+            if (Arrays.stream(RomanOnes).toList().contains(romanNumber.substring(currentDigit, currentDigit + 1))) {
+                int digit = Arrays.stream(RomanOnes).toList().indexOf(romanNumber.substring(currentDigit, currentDigit + 1));
+                arabicNumber += getOnes(romanNumber, currentDigit, RomanDigits[digit]);
             }
 
             if (romanNumber.toUpperCase().charAt(currentDigit) == V.value().charAt(0)) {
-                unit += 5;
+                arabicNumber += 5;
             }
         }
+        return arabicNumber;
+    }
+
+    private int getOnes(String romanNumber, int currentDigit, Digits digit) {
+
+        int sum = 0;
+
+        int nextDigit = currentDigit + 1;
+        if (nextDigit == romanNumber.length() ||
+            isNextDigitEqualOrLess(romanNumber.toUpperCase().charAt(currentDigit), romanNumber.toUpperCase().charAt(nextDigit))) {
+            sum = 1;
+        } else {
+            sum = -1;
+        }
+
+        return sum * (RomanDigits[digit.position()]).arabicValue();
+    }
 
-        return unit;
+    private boolean isNextDigitEqualOrLess(char currentDigit, char nextDigit) {
+        int currentArabicDigit = parseRomanToArabic(currentDigit + "");
+        int nextArabicDigit = parseRomanToArabic(nextDigit + "");
```

Comment:
Is it okay to use another method like I did or the code became high copled? @chicofaccin

Reply:
I think I didn't understand 😅

Reply:
We can talk about it in the next meeting as well